Mathlib/Data/Set/Finite/Basic.lean

```diff
@@ -688,6 +688,22 @@ theorem Finite.dinduction_on {C : ∀ s : Set α, s.Finite → Prop} (s : Set α
     Finite.induction_on h (fun _ => H0) fun has hs ih _ => H1 has hs (ih _)
   this h
 
+theorem setOf_nodup_perm (l : List α) (H : l.Nodup) [DecidableEq α]
+    : l.finite_toSet.toFinset.toList.Perm l := by
+  rw [List.perm_iff_count]
+  intro a
+  by_cases h : a ∈ l
+  · rw [l.count_eq_one_of_mem H h,
+      l.finite_toSet.toFinset.toList.count_eq_one_of_mem (Finset.nodup_toList _) (by simp_all)]
+  · rw [l.count_eq_zero_of_not_mem h,
+      l.finite_toSet.toFinset.toList.count_eq_zero_of_not_mem (by simp_all)]
+
+theorem setOf_eq_singleton_iff_of_nodup (l : List α) (H : l.Nodup) {a : α} [DecidableEq α]
+    : { x | x ∈ l } = {a} ↔ l = [a] := by
```

Comment:
The theorem name doesn't indicate at all that a list is involved. Maybe `setOf_mem_list_eq_singleton` optionally followed by `_iff` and/or `_of_nodup`.